src/com/softserve/academy/edu1/hw5/taskHW3/Car.java

```diff
@@ -0,0 +1,113 @@
+package softserve.academy.edu1.hw5.taskHW3;
+
+import java.util.*;
+
+public class Car implements Comparable<Car> {
+    static final Scanner SCANNER = new Scanner(System.in);
+
+    private String type;
+    private int yearOfProduction;
+    private double engineCapacity;
+
+    public Car(String type, int yearOfProduction, double engineCapacity) {
+        this.type = type;
+        this.yearOfProduction = yearOfProduction;
+        this.engineCapacity = engineCapacity;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public int getYearOfProduction() {
+        return yearOfProduction;
+    }
+
+    public double getEngineCapacity() {
+        return engineCapacity;
+    }
+
+    public static int getMaxYear (Car[] cars) {
```

Comment:
Almost all methods contain the parameter Car[] cars. In real life, this is a reason to think about separating this class into two: class Car and class CarService that manages collections of car objects